examples/gno.land/p/demo/avl/list/list.gno

```diff
@@ -0,0 +1,337 @@
+	}
+
+	if index < 0 || index > size {
+		return false
+	}
+
+	// If setting at the end (append case)
+	if index == size {
+		l.Append(value)
+		return true
+	}
+
+	// Get the key at the specified index
+	key, _ := l.tree.GetByIndex(index)
+	if key == "" {
+		return false
+	}
+
+	// Update the value at the existing key
+	l.tree.Set(key, value)
+	return true
+}
+
+// Delete removes the element at the specified index.
+// Returns the deleted value and true if successful, nil and false otherwise.
+//
+// Example:
+//
+//	l := list.New()
+//	l.Append(1, 2, 3)
+//
+//	val, ok := l.Delete(1)
+//	println(val, ok)  // Output: 2 true
+//	println(l.Len())  // Output: 2
+//
+//	val, ok = l.Delete(-1)
+//	println(val, ok)  // Output: nil false
+func (l *List) Delete(index int) (interface{}, bool) {
+	size := l.tree.Size()
+	// Always return nil, false for empty list
+	if size == 0 {
+		return nil, false
+	}
+
+	if index < 0 || index >= size {
+		return nil, false
+	}
+
+	key, value := l.tree.GetByIndex(index)
+	if key == "" {
+		return nil, false
+	}
+
+	l.tree.Remove(key)
+	return value, true
+}
+
+// Slice returns a slice of values from startIndex (inclusive) to endIndex (exclusive).
+// Returns nil if the range is invalid.
+//
+// Example:
+//
+//	l := list.New()
+//	l.Append(1, 2, 3, 4, 5)
+//
+//	println(l.Slice(1, 4))   // Output: [2 3 4]
+//	println(l.Slice(-1, 2))  // Output: [1 2]
+//	println(l.Slice(3, 999)) // Output: [4 5]
+//	println(l.Slice(3, 2))   // Output: nil
+func (l *List) Slice(startIndex, endIndex int) []interface{} {
+	size := l.tree.Size()
+
+	// Normalize bounds
+	if startIndex < 0 {
+		startIndex = 0
+	}
+	if endIndex > size {
+		endIndex = size
+	}
+	if startIndex >= endIndex {
+		return nil
+	}
+
+	count := endIndex - startIndex
+	result := make([]interface{}, count)
+
+	// Fill the slice from end to start to maintain correct order
+	i := count - 1
+	l.tree.IterateByOffset(size-endIndex, count, func(_ string, value interface{}) bool {
+		result[i] = value
+		i--
+		return false
+	})
+	return result
+}
+
+// ForEach iterates through all elements in the list.
+// The callback function receives the index and value of each element.
+// Return true from the callback to stop iteration.
+//
+// Example:
+//
+//	l := list.New()
+//	l.Append(1, 2, 3)
+//
+//	sum := 0
+//	l.ForEach(func(index int, value interface{}) bool {
+//	    sum += value.(int)
+//	    return false  // continue iteration
+//	})
+//	println(sum)  // Output: 6
+//
+//	// Early stop
+//	l.ForEach(func(index int, value interface{}) bool {
+//	    if value.(int) > 2 {
+//	        return true  // stop iteration
+//	    }
+//	    return false
+//	})
+func (l *List) ForEach(fn func(index int, value interface{}) bool) {
+	size := l.tree.Size()
+	values := make([]interface{}, size)
+
+	// First collect values in reverse order
+	i := size - 1
+	l.tree.IterateByOffset(0, size, func(_ string, value interface{}) bool {
+		values[i] = value
+		i--
+		return false
+	})
```

Comment:
If reverse here is necessary, you can use ReverseIterateByOffset and call fn() directly